library/src/main/java/com/eggheadgames/realmassethelper/OsUtil.java

```diff
@@ -21,7 +21,10 @@ public String loadDatabaseToLocalStorage(Context context, String databaseFolder,
 
         File file = new File(generateDatabaseFileName(context, databaseName));
         if (file.exists()) {
-            file.delete();
+            boolean delete = file.delete();
+            if (!delete) {
+                throw new RuntimeException("Can not remove old database");
```

Comment:
As far as I can see this method throws Exception if it can not write to the file. So I can assume that throw RuntimeException("Can not remove old database") here will not create a problem, because that case it will be thrown below.

Reply:
Yes, how about we move the if & delete into the existing try/catch below?